### ISSUE TITLE
Accessibility : RGAA compatibility

### ISSUE THREAD

**Issue by tfrancart** [sparna-git/Sparnatural#431]
Compatibility with [RGAA](https://design.numerique.gouv.fr/accessibilite-numerique/RGAA/) and more generally "[critères qualité des sites de l’Etat](https://design.numerique.gouv.fr/outils/checklist-pidila/)".

**Comment by simongray** [comment]
I really think this query builder looks amazing, but it is a shame that any website in the EU that chooses to use it will technically be in violation of the accessibility directive (and associated national laws).

A bit of keyboard access would certainly be welcome for any type of user, not just for people who can't use a pointing device.

**Comment by tfrancart** [comment]
Thanks for your comment ! I am definitely not an accessibility expert, concretely what would your suggestions imply ? can you give an HTML snippet of Sparnatural with added attributes to deal with keyboard access for example ?

**Comment by simongray** [comment]
It's not really something you can add in a snippet, unfortunately.

Once the developer deviates from the built-in widgets, a lot of things have to be declared as aria attributes and custom event handlers have to be written to deal with keyboard input

I have used the WCAG before since it includes examples for different classes of widgets, e.g  for a combobox you can use https://w3c.github.io/aria/#combobox as a guideline.

Basically, to be compliant you need to:

1. **not** remove focus indicators using CSS (this is a big mistake)
2. add keyboard event handlers
3. add ARIA and other attributes to the relevant HTML structures

**Comment by simongray** [comment]
Usually, the keyboard access is solved by adding tabindex attributes to widgets so that you can tab between them, using keyboard arrows and enter to manipulate the widget in focus.
